server/src/main/java/org/apache/iotdb/db/qp/strategy/PhysicalGenerator.java

```diff
@@ -416,11 +418,12 @@ private void generateDataTypes(QueryPlan queryPlan) throws PathException {
   }
 
   private void deduplicate(QueryPlan queryPlan) {
+    if (queryPlan.isGroupByDevice()) {
+      return;
```

Comment:
Add a comment to explain this like "The deduplication of a GroupByDevice query is done in the dataset".

Reply:
done, thanks for your advice